packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -270,7 +273,12 @@ export function getStylesDeclarations(
 		const cssProperty = rule.key.startsWith( '--' )
 			? rule.key
 			: kebabCase( rule.key );
-		output.push( `${ cssProperty }: ${ compileStyleValue( rule.value ) }` );
+		let ruleValue = rule.value;
+		if ( typeof ruleValue !== 'string' && ruleValue?.ref ) {
+			const refPath = ruleValue.ref.split( '.' );
+			ruleValue = get( tree, refPath );
```

Comment:
I _think_ this will mutate the `rule` variable, which is OK but probably better if we can avoid it. I know this was already an issue before your PR. Maybe an extra function is a good idea, but making it a bit bigger to encapsulate the logic for both the property and the value? That might be better in a follow up PR anyway...

Reply:
What would mutate it? This:
```
const rule = { value: 1 };
let ruleValue = rule.value;
console.log(ruleValue);
ruleValue = 2;
console.log( rule )
console.log( ruleValue )
```
results in:
```
1
Object { value: 1 }
2
```